lib/signaling/v2/peerconnectionmanager.js

```diff
@@ -247,11 +247,12 @@ class PeerConnectionManager extends QueueingEventEmitter {
    */
   setTrackSenders(trackSenders) {
     const dataTrackSenders = new Set(trackSenders.filter(trackSender => trackSender.kind === 'data'));
-    const mediaTrackSenders = new Set(trackSenders.filter(trackSender => trackSender.kind === 'audio' || trackSender.kind === 'video'));
 
-    if (this._dummyAudioTrackSender) {
-      mediaTrackSenders.add(this._dummyAudioTrackSender);
-    }
+    const mediaTrackSenders = trackSenders.reduce((mediaTrackSenders, trackSender) =>
```

Comment:
@markandrus I think this can be simplified to:
```js
trackSenders = (this._dummyAudioTrackSender
  ? [this._dummyAudioTrackSender]
  : []).concat(trackSenders);

const mediaTrackSenders  = new Set(trackSenders.filter(trackSender => {
  return trackSender.kind === 'audio' || trackSender.kind === 'video';
}));
```
WDYT?

Reply:
I'd rather we not reassign `trackSenders`, so we could do this instead

```js
const mediaTrackSenders = new Set([this._dummyAudioTrackSender]
  .concat(mediaTrackSenders)
  .filter(trackSender => trackSender && (trackSender.kind === 'audio' || trackSender.kind === 'video')));
```